openedx/core/lib/api/authentication.py

```diff
@@ -20,22 +21,38 @@
 log = logging.getLogger(__name__)
 
 
-class OAuth2AuthenticationAllowInactiveUser(OAuth2Authentication):
+class OAuth2AuthenticationDeprecated(OAuth2Authentication):
+    """
+    This child class was added to add new_relic metrics to OAuth2Authentication. This should be very temporary.
+    """
+
+    def authenticate(self, request):
+        """
+        Returns two-tuple of (user, token) if access token authentication
+        succeeds, raises an AuthenticationFailed (HTTP 401) if authentication
+        fails or None if the user did not try to authenticate using an access
+        token.
+        """
+        set_custom_metric("OAuth2AuthenticationCalled", True)
+        output = super(OAuth2AuthenticationDeprecated, self).authenticate(request)
+        set_custom_metric("OAuth2AuthenticationSuccess", True)
+        return output
+
+
+class OAuth2AuthenticationAllowInactiveUser(OAuth2AuthenticationDeprecated):
     """
     This is a temporary workaround while the is_active field on the user is coupled
     with whether or not the user has verified ownership of their claimed email address.
     Once is_active is decoupled from verified_email, we will no longer need this
     class override.
-
     But until then, this authentication class ensures that the user is logged in,
     but does not require that their account "is_active".
-
     This class can be used for an OAuth2-accessible endpoint that allows users to access
     that endpoint without having their email verified.  For example, this is used
     for mobile endpoints.
     """
 
-    def authenticate(self, *args, **kwargs):
+    def authenticate(self, request):
```

Comment:
Seems fine, but do we know this won't break anything?

Reply:
It shouldn't. DRF standardizes this to only take request as an input, so there should not be another function that has more inputs.